cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -378,7 +374,14 @@ private String rebalanceStateConditionType(KafkaRebalanceStatus status) {
             return updateStatus(reconciliation, kafkaRebalance, status, null).compose(i -> Future.succeededFuture());
         }
 
-        RebalanceOptions.RebalanceOptionsBuilder rebalanceOptionsBuilder = convertRebalanceSpecToRebalanceOptions(kafkaRebalance.getSpec(), usingJbodStorage);
+        if (kafkaRebalance.getSpec().isRebalanceDisk() && !usingJbodStorage) {
+            String error = "Cannot set rebalanceDisk=true for Kafka clusters with a non-JBOD storage config. " +
+                "Intra-broker balancing only applies to Kafka deployments that use JBOD storage with multiple disks.";
+            LOGGER.errorCr(reconciliation, "Status updated to [NotReady] due to error: {}", new IllegalArgumentException(error).getMessage());
+            return updateStatus(reconciliation, kafkaRebalance, new KafkaRebalanceStatus(), new IllegalArgumentException(error)).mapEmpty();
```

Comment:
Should we throw `InvalidResourceException`? I do not know much the Rebalance operator. But in others we would normally use `InvalidResourceException`.

Reply:
Yes I agree. It's referring to the `KafkaRebalance` resource which is wrong in this case so the `InvalidResourceException` makes more sense to me.

Reply:
Agreed

Reply:
Updated